File: mobile/src/main/java/com/pdyjak/powerampwear/onboarding/OnboardingViewPagerAdapter.java
package com.pdyjak.powerampwear.onboarding;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

class OnboardingViewPagerAdapter extends FragmentPagerAdapter {
    static final int PAGES_COUNT = 3;

    OnboardingViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new WelcomeFragment();

            case 1:
                return new InstallationVerificationFragment();

            case 2:
                return new AlmostReadyFragment();
        }
        return null;
    }

    @Override
    public int getCount() {
        return PAGES_COUNT;
    }
}

File: mobile/src/main/java/com/pdyjak/powerampwear/onboarding/OnboardingViewPagerAdapter.kt
package com.pdyjak.powerampwear.onboarding

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

internal class OnboardingViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    companion object {
        const val PAGES_COUNT = 3
    }

    override fun getItem(position: Int): Fragment? {
        when (position) {
            0 -> return WelcomeFragment()
            1 -> return InstallationVerificationFragment()
            2 -> return AlmostReadyFragment()
        }
        return null
    }

    override fun getCount(): Int {
        return PAGES_COUNT
    }
}
